django/db/models/sql/compiler.py

```diff
@@ -487,9 +487,10 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
         refcounts_before = self.query.alias_refcount.copy()
         try:
             extra_select, order_by, group_by = self.pre_sql_setup()
-            for_update_part = None
             # Is a LIMIT/OFFSET clause needed?
             with_limit_offset = with_limits and (self.query.high_mark is not None or self.query.low_mark)
+            # Calculate the update part - it might alter the joins
+            for_update_part = self.get_for_update_part(with_limit_offset)
```

Comment:
It's not used only for combined queries so we should call it only in `not combinator` branch.

Reply:
Sorry I couldn't catch that. Are you saying we should move `for_update_part` inside the `else` clause? But that's not possible since it is also used outside the else-block scope.

Reply:
We should leave here `for_update_part = None` (in the same line as it was previously) and calculate it only in `not combinator` branch, because it's not used for combined queries.

Reply:
Done, thanks.